index.php

```diff
@@ -4,29 +4,53 @@
 require_once 'src/SeedsViewer.php';
 require_once 'src/Seed.php';
 
+
+
 $db = new PDO('mysql:host=db; dbname=seed_vault', 'root', 'password');
 $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC); 
 
+$query = $db->prepare('SELECT `seeds`.`id`, `seeds`.`name`, `plant_family`.`common_name` AS "Plant Family",
+`seeds`.`species`, `seeds`. `image`, `seeds`.`description`
+FROM `seeds`
+	INNER JOIN `plant_family`
+		ON `seeds`.`genus` = `plant_family`.`id`');
+
+$query->execute();
+
+$result = $query->fetchAll();
+
 $seedsModel = new SeedsModel($db); 
 $seeds = $seedsModel->getAllSeeds();
 
+
 if (
     isset($_POST['name']) &&
-    isset($_POST['family_name']) &&
+    isset($_POST['genus']) &&
     isset($_POST['species']) &&
     isset($_POST['image']) &&
     isset($_POST['description'])
 ) {
-    // Saving the data into variables
     $name = $_POST['name'];
-    $family_name = $_POST['family_name'];
+    $genus = $_POST['genus'];
     $species = $_POST['species'];
-    $species = $_POST['image'];
-    $species = $_POST['description'];
-
-    $db = connectToDb();
+    $image = $_POST['image'];
+    $description = $_POST['description'];
+
+    if (strlen($name) == 0 || strlen($name) > 31) {
+        echo "Please fill in seed name that is between 1 to 30 characters!";
+    } else if (intval($genus) == 0 || intval($genus) > 6) {
+        echo "Please select a Plant Value from the dropdown menu!";
+    } else if (strlen($species) == 0 || strlen($species) > 41) {
+        echo "Please fill in species name that is between 1 to 40 characters!";
+    } else if (strlen($image) == 0 || strlen($image) > 2000) {
+        echo  "Please fill in a URL of an image that is between 1 to 2000 characters!";
+    } else if (strlen($description) == 0 || strlen($description) > 2000) {
+        echo "Please write a description that is between 1 to 500 characters!";
+    } else {
```

Comment:
Validation and logic here looks perfect, but it would be nice to display the error messages in or next to the form.

Technically speaking as well, echoing them out up here means they are being displayed outside the body tag, which is invalid HTML.

Have a think about a way you can keep the if statements up here, but display the errors further down the page